app/controllers/api/v2/vcs_clone_controller.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    class VcsCloneController < ::Api::V2::BaseController
+      include ::ForemanAnsible::ProxyAPI
+
+      rescue_from ActionController::ParameterMissing do |e|
+        render json: { 'error' => e.message }, status: :bad_request
+      end
+
+      rescue_from Git::GitExecuteError do |_e|
+        head :internal_server_error
+      end
+
+      api :GET, '/vcs_clone/get_repo_info', N_('Returns information about the repo')
+      param :vcs_url, String, N_('Url of the repo'), :required => true
+      error 400, :desc => N_('Parameter unfulfilled')
+      error 500, :desc => N_('Git error')
+      def repo_information
+        vcs_url = params.require(:vcs_url)
+        remote = Git.ls_remote(vcs_url).slice('head', 'branches', 'tags')
+        remote['vcs_url'] = vcs_url
+        render json: remote
+      end
+
+      api :GET, '/vcs_clone/get_installed_roles', N_('Returns an array of roles installed on the provided proxy')
+      formats ['json']
+      param :smart_proxy, Array, N_('Name of the SmartProxy'), :required => true
+      error 400, :desc => N_('Parameter unfulfilled')
+      error 500, :desc => N_('Internal server error')
+      def installed_roles
+        smart_proxy = params.require(:smart_proxy)
+        ansible_proxy = SmartProxy.find_by(name: smart_proxy)
+        proxy_api = find_proxy_api(ansible_proxy)
+        installed = proxy_api.list_installed
+        render json: installed
+      rescue Foreman::Exception
+        head :internal_server_error
+      end
+
+      api :POST, '/vcs_clone/install', N_('Launches a task to install the provided role')
+      formats ['json']
+      param :repo_info, Hash, :desc => N_('Dictionary containing info about the role to be installed') do
+        param :vcs_url, String, :desc => N_('Url of the repo'), :required => true
+        param :name, String, :desc => N_('Name of the repo'), :required => true
+        param :ref, String, :desc => N_('Branch / Tag / Commit reference'), :required => true
+        param :update, String, :desc => N_('Whether an existing role with the provided name should be updated'), :default => false
+      end
+      param :smart_proxy, Array, N_('Array of SmartProxies the role should get installed to')
```

Comment:
Why is this an array, but only the first ID gets used?

Reply:
This has been indirectly resolved by passing the proxy-id as a url parameter.